### DISCUSSION TITLE
SEB Windows keyboard layout does not display all selectable keyboard language layouts

### DISCUSSION THREAD

**Discussion by nvarghese-rm** [SafeExamBrowser/seb-win-refactoring#128] [Q&A]
Hi

In SEB Windows v3.2 as well as v2.3, when you click to select the keyboard layout, not all of the keyboard layout options available in Windows are displayed to the user in SEB. For example, language layouts for Chinese and Japanese are not available as shown in the attached image.

![Japanese Keyboard](https://user-images.githubusercontent.com/87759879/127072251-eea6cd57-d749-44e5-91e4-11c67734e4cd.png)

Is this deliberate or is it a bug? Are there any plans to address this?
Thanks

**Comment by dbuechel** [comment]
You'd need to add a new language, only adding an input method won't work (it at least used to be that way, it is now a while since we last revisited this topic). See also #18, specifically [this comment](https://github.com/SafeExamBrowser/seb-win-refactoring/issues/18#issuecomment-886429556).

**Reply by nvarghese-rm** [reply]
Thanks, that was very useful. 
I already had the languages added but as soon as I set Kiosk Mode as Disable Explorer Shell, the Japanese and Chinese options appeared in the keyboard layout menu for both v3.2 and v2.3 .  But the 'candidate character options' did not appear. It would only appear if I disabled Kiosk mode entirely.  

So not entirely working as expected if we can't run in Kiosk mode.

**Reply by dbuechel** [reply]
Okay, then we'll have a look at the issue in context of #18. We can't however guarantee that we'll be able to fix it, as the IME never used to work reliably within SEB's kiosk modes.

**Reply by dbuechel** [reply]
@nvarghese-rm If you have any further aspects or info, feel free to add them to #18.

**Reply by nvarghese-rm** [reply]
Thanks very much  - will defer to #18 for any other discussion. Had a few more things to say about Japanese language.
